fsdev/db.cpp

```diff
@@ -143,38 +131,35 @@ bool InsertTrack(sql::Connection * connection, DIRENT * d, string & nspace, Logg
 		return rv;
 	}
 
-	const int bsize = 1024;
-	char buffer[bsize] = { 0 };
+	char buffer[BSIZE] = { 0 };
 
 	Track track;
 
-	while (fgets(buffer, bsize, p) != nullptr)
+	while (fgets(buffer, BSIZE, p) != nullptr)
 	{
 		string b(buffer);
 		track.SetTag(b);
-		memset(buffer, 0, bsize);
+		memset(buffer, 0, BSIZE);
 	}
 
 	pclose(p);
 
+	size_t n = sizeof(track_column_names) / sizeof(string);
+	sql::PreparedStatement * stmt = nullptr;	
+	string s = "insert into tracks " + query_columns + parameter_columns + ";";
+
 	try
 	{
-		size_t n = sizeof(track_column_names) / sizeof(string);
-		sql::PreparedStatement * stmt = nullptr;	
-		string s = "insert into tracks " + query_columns + parameter_columns + ";";
 
 		stmt = connection->prepareStatement(s.c_str());
-		if (stmt == nullptr)
-		{
-			cerr << "prepareStatement() failed" << endl;
-			return rv;
+		if (stmt == nullptr) {
+			throw LOG2(_log_, "prepareStatement() failed", LogLevel::FATAL);
 		}
 		stmt->setString(1, to_string(d->up));
 		stmt->setString(2, d->name);
 		stmt->setString(3, nspace);
 
-		for (size_t i = 3; i < n; i++)
-		{
+		for (size_t i = 3; i < n; i++) {
```

Comment:
Note above the magical 3, there are three setStrings. These three special ones come from the "file system structure" being built by this program. The remaining fields come from ffprobe. Adding comments now.